### PR TITLE
Fix: multiple: true for outputs including unit test

### LINKED ISSUE #638: [BUG] A VDSL3 module does not handle `multiple: true` outputs properly
### What happened?

Setting `multiple: true` on an output generates an error:

```
ERROR ~ No such variable: destPath
```

This error is generated in the `publishStatesProc` process. The issues arise in the `publishStates.nf` source that is included in the generated module.

### Steps to reproduce

See `multiple_output` unit test in under `src/test/resources/testnexrtflowvdsl3/src`.

### Expected behavior

The directory should be created.

### Relevant log output

_No response_

### Version

Viash 0.8.0 and above

### Possible solution

The solution is to fix line 187 in `publishSteates.nf`.

### Confirmation

- [X] I have searched the existing issues to make sure this is not a duplicate.
- [X] I have provided clear and concise information about the bug.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tverbeiren**
## Describe your changes

* `NextflowPlatform`: Fix publishing state for output arguments with `multiple: true`. 

* `Executable`: Check whether a multiple output file argument contains a wildcard.

## Related issue(s)

Closes #638 

## Type of Change

- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New functionality (non-breaking change which adds functionality)
- [ ] Major change (non-breaking change which modifies existing functionality)
- [ ] Minor change (non-breaking change which does not modify existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] This change requires a documentation update

## Checklist

Requirements:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc.
- [x] I have performed a self-review of my code by checking the "Changed Files" tab.
- [x] My code follows the code style of this project.

Tests:

- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

Documentation:

- [x] Proposed changes are described in the CHANGELOG.md.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by rcannood — APPROVED** [review]
LGTM

**Review by rcannood — CHANGES_REQUESTED** [review]
Wait!

File: src/main/scala/io/viash/wrapper/BashWrapper.scala [hunk 1]
```diff
@@ -456,7 +461,12 @@ object BashWrapper {
       } else {
         "\n# check whether required files exist\n" +
           files.map { param =>
-            if (param.multiple) {
+            if (!param.multiple) {
+              s"""if [ ! -z "$$${param.VIASH_PAR}" ] && [ ! -e "$$${param.VIASH_PAR}" ]; then
+                |  ViashError "$direction file '$$${param.VIASH_PAR}' does not exist."
+                |  exit 1
+                |fi""".stripMargin
```
Comment (Grifs) [inline src/main/scala/io/viash/wrapper/BashWrapper.scala:468]:
```suggestion
              s"""if [ ! -z "$$${param.VIASH_PAR}" ] && [ ! -e "$$${param.VIASH_PAR}" ]; then
                 |  ViashError "$direction file '$$${param.VIASH_PAR}' does not exist."
                 |  exit 1
                 |fi""".stripMargin
```

**Review by Grifs — COMMENTED** [review]
all seems pretty reasonable. Found a minor indentation fault.

File: src/main/scala/io/viash/wrapper/BashWrapper.scala (hunk 1 above)
Comment (rcannood) [reply src/main/scala/io/viash/wrapper/BashWrapper.scala:468]: Thx!

**Review by rcannood — APPROVED** [review]

File: src/main/scala/io/viash/functionality/arguments/FileArgument.scala [hunk 2]
```diff
@@ -121,7 +121,19 @@ case class FileArgument(
...
+  @description(
+    """Allow for multiple values (`false` by default).
+      |
+      |For input arguments, this will be treated as a list of values. For example, values
+      |can be passed using the delimiter `--foo=1:2:3` or by providing the same argument 
+      |multiple times `--foo 1 --foo 2`. You can use a custom delimiter by using the 
+      |[`multiple_sep`](#multiple_sep) property.
+      |
+      |For output file arguments, the passed value needs to contain a wildcard. For example,
+      |'--foo=foo_*.txt' will be treated as a list of files that match the pattern.
```
Comment (rcannood) [inline src/main/scala/io/viash/functionality/arguments/FileArgument.scala:133]:
```suggestion
      |For output file arguments, the passed value needs to contain a wildcard. For example,
      |`--foo 'foo_*.txt'` will be treated as a list of files that match the pattern. Note that in Bash,
      | the wildcard will need to be in quotes (`"foo_*.txt"` or `'foo_*.txt'`) or else Bash will
      | automatically attempt to expand the expression.
```

Comment (tverbeiren) [reply src/main/scala/io/viash/functionality/arguments/FileArgument.scala:133]: +1

**Comment by tverbeiren** [comment]
LGTM!

**Pull request merged**